lib/kotlin/build.gradle.kts

```
@@ -34,15 +36,15 @@ dependencies {
     testImplementation(kotlin("test"))
 }
 
-kotlin {
-    jvmToolchain {
-        (this as JavaToolchainSpec).languageVersion.set(JavaLanguageVersion.of(8))
-    }
+java {
+    sourceCompatibility = JavaVersion.VERSION_1_8
+    targetCompatibility = JavaVersion.VERSION_1_8
 }
 
-tasks.withType<org.jetbrains.kotlin.gradle.tasks.KotlinCompile> {
+tasks.withType<KotlinCompile> {
     compilerOptions {
-        jvmTarget = org.jetbrains.kotlin.gradle.dsl.JvmTarget.JVM_1_8
+        jvmTarget = JvmTarget.JVM_1_8
+        freeCompilerArgs = listOf("-Xjdk-release=1.8")
```

The [docs](https://kotlinlang.org/docs/compiler-reference.html#xjdk-release-version) say that setting `-Xjdk-release=version` automatically sets the `-jvm-target version` as well. I don't think it hurts to have both, though, so long as they are the same.

So, I think this part is okay. But the examples I saw appended to the list:

```suggestion
        freeCompilerArgs += listOf("-Xjdk-release=1.8")
```

---

Added it for extra clarification, I dont mind removing it either, let me know.

As for the `+=`, the blog modifies the `freeCompilerArgs` afterwards, whereas we modify it here during the initial configuration. So `+=` doesn't work here.